src/utils/dropNilAndEmptyString.ts

```diff
@@ -0,0 +1,24 @@
+import isNil from 'lodash/isNil';
+import isObject from 'lodash/isObject';
+
+const dropNilAndEmptyString = (d: object): object => {
+  return Object.entries(d).reduce((acc, [k, v]) => {
+    if (isNil(v) || v === '') {
+      return acc;
+    } else if (Array.isArray(v)) {
+      return {
+        ...acc,
+        [k]: v.map((item) =>
+          isObject(item) ? dropNilAndEmptyString(item) : item
+        ),
+      };
+    } else {
+      return {
+        ...acc,
+        [k]: isObject(v) ? dropNilAndEmptyString(v) : v,
+      };
+    }
+  }, {});
+};
+
+export default dropNilAndEmptyString;
```

Comment:
where is this used?

anyway, it could be maybe simplified, if those objects don't contain functions

```
const dropNilAndEmptyString = (d) => {
  return JSON.parse(
    JSON.stringify(d, (k, v) => {
      if (v != null && v !== '') {
        return v;
      }
      return undefined;
    })
  );
};
```

Reply:
see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify